intro.adoc

```diff
@@ -1,18 +1,19 @@
 
 
 This specification defines non-ISA extension <<dmsext, Debug Module Security Extension (non-ISA extension)>> and ISA extension <<Sdsec, Sdsec (ISA extension)>> to address the above security issues in the current _The RISC-V Debug Specification_ cite:[dbgspec] and trace specifications cite:[etrace] cite:[ntrace]. 
 
 A summary of the changes introduced by _The RISC-V External Debug Security Specification_ follows.:
 
-    - *Per-Hart Debug Control:* Introduce per-hart control knobs to control whether external debug is allowed in M-mode and/or supervisor domains cite:[smmtt]
-    - *Debug Mode:* External debugger can only halt the hart and enter debug mode when debug is allowed in current privilege mode; all operations are executed with <<dbgaccpriv, debug access privilege>> instead of M-mode privilege
-    - *Memory Access:* Memory access from a hart’s point of view using a Program Buffer or the Abstract Command must be checked by the hart's memory protection mechanisms as if the hart is running at <<dbgaccpriv, debug access privilege>>; memory access from Debug Module using System Bus Access block without involving a hart must be checked by system memory protection mechanism, such as IOPMP or WorldGuard
-    - *Register Access:* Register access using Program Buffer or the Abstract Command works as if the hart is running in <<dbgaccpriv, debug access privilege>> instead of M-mode privilege
-    - *Triggers:* Triggers (with action=1) can only fire or match when external debug is allowed in current privilege
-    - *Per-Hart Trace Control:* Introduce per-hart knobs to control whether tracing is allowed in M-mode and supervisor domains
+    - *Per-Hart Debug Control:* Introduce per-hart states to control whether external debug is allowed in M-mode and/or supervisor domains cite:[smmtt].
+    - *Per-Hart Trace Control:* Introduce per-hart states to control whether tracing is allowed in M-mode and/or supervisor domains.
+    - *Non-secure debug:* Add a non-secure debug state to relax security constraints.
+    - *Debug Mode entry:* External debugger can only halt the hart and enter debug mode when debug is allowed in current privilege mode; all operations are executed with <<dbgaccpriv, debug access privilege>> instead of M-mode privilege.
+    - *Memory Access:* Memory access from a hart’s point of view using a Program Buffer or the Abstract Command must be checked by the hart's memory protection mechanisms as if the hart is running at <<dbgaccpriv, debug access privilege>>; memory access from Debug Module using System Bus Access block without involving a hart must be checked by system memory protection mechanism, such as IOPMP or WorldGuard.
```

Comment:
> a Program Buffer or the Abstract Command

I would say "the Program Buffer or an Abstract Command"

Reply:
> from Debug Module

I would say "from the Debug Module"

Reply:
It took me a few minutes to understand the part after the semicolon.  I think that it should be:
```suggestion
    - *Memory Access:* Memory access from a hart’s point of view using the Program Buffer or an Abstract Command must be checked by the hart's memory protection mechanisms as if the hart is running at <<dbgaccpriv, debug access privilege>>; memory access from the Debug Module using System Bus Access must be checked by a system memory protection mechanism, such as IOPMP or WorldGuard.
```
SBA is defined to not use a hart.  Either the above or "memory access from the Debug Module using System Bus Access without involving a hart..."  My main confusion was that "block" can be either a verb or a noun and SBA is architecturally more of a feature than a block so my brain tried to parse it as a verb.

Reply:
Sorry for the confusion 😂. Will take your suggestion